src/gromo/conv2d_growing_module.py

```diff
@@ -246,6 +284,118 @@ def compute_m_update(
             self.input.shape[0],
         )
 
+    def compute_m_prev_update(
+        self, desired_activation: torch.Tensor | None = None
+    ) -> tuple[torch.Tensor, int]:
+        """
+        Compute the update of the tensor M_{-2} := B[-2]^T dA . # TODO: correct this formula
+
+        Parameters
+        ----------
+        desired_activation: torch.Tensor | None
+            desired variation direction of the output  of the layer
+
+        Returns
+        -------
+        torch.Tensor
+            update of the tensor M_{-2}
+        int
+            number of samples used to compute the update
+        """
+        if desired_activation is None:
+            desired_activation = self.pre_activity.grad
+        desired_activation = desired_activation.flatten(start_dim=-2)
+
+        if self.previous_module is None:
+            raise ValueError(
+                f"No previous module for {self.name}. Thus M_{-2} is not defined."
+            )
+        elif isinstance(self.previous_module, LinearGrowingModule):
+            raise NotImplementedError("TODO: implement this")
+        elif isinstance(self.previous_module, LinearAdditionGrowingModule):
+            raise NotImplementedError("TODO: implement this")
+        elif isinstance(self.previous_module, Conv2dGrowingModule):
+            return (
+                torch.einsum(
+                    "ixab, icx -> bca",
+                    self.prev_masked_unfolded_activation,
+                    desired_activation,
+                ),
+                self.input.shape[0],
+            )
+        elif isinstance(self.previous_module, Conv2dAdditionGrowingModule):
+            raise NotImplementedError("TODO: implement this")
+        else:
+            raise NotImplementedError(
+                f"The computation of M_{-2} is not implemented yet "
+                f"for {type(self.previous_module)} as previous module."
+            )
+
+    def compute_cross_covariance_update(self) -> tuple[torch.Tensor, int]:
+        """
+        Compute the update of the tensor P := B[-2]^T B[-1] .# TODO: correct this formula
```

Comment:
A point to discuss later: is it interesting to use shrinkage estimator for cross-covariance, to ensure positive definiteness even with small mini batch?